docs/request.response.structures.md

```diff
@@ -0,0 +1,179 @@
+
+
+## Outgoing requests from presto
+
+* The outgoing reponse request includes the payloads and parameters from the incoming request
+* items in the response must include their id, so that the caller can match and lookup individual-level properties from incoming request. 
+* status code is present at top level so it can be checked to know how to parse the rest of the payload (error etc)
+
+Example PROPOSED output structure
+```
+{
+    request_id:122,              
+    model_name:"classycat__Model",       # model to be used 
+    model_parameters: {
+        "event_type": "classify",
+        "schema_id": "4a026b82-4a16-440d-aed7-bec07af12205",
+    },
+    input_items:[
+        {
+            id:"11",
+            text:"this is some text to classify", 
+            workspace_id:"timpani_meedan", 
+            target_state:"classified", 
+            language:"en"
+
+        },
+        {
+            id:"12",
+            text:"Este é um texto para classificar", 
+            workspace_id:"timpani_meedan", 
+            target_state:"classified", 
+            language:"pt"
+        },
+    ],
+    callback_url: "http://conductor/batch_reply/",
+
+    # elements below here are added in response
+
+    "retry_count": 0
+    "status_code": 200
+    "status_message":"success"
+    "results": [
+            {
+                "id": "11",     #  this id can be used to look up params from input
+                "text": "this is some text to classify",   # text is redundant with input, but helpful for model-level debugging
+                "labels": [
+                    "Testing",
+                    "Classification"
+                ]
+            },
+            {
+                "id": "12",
+                "text": "Este é um texto para classificar",
+                "labels": [
+                    "Testing",
+                    "Classification"
+                ]
+            }
+        ]
+    },
+
+    ```
+
+Example PROPOSED output object structure (YAKE)
+{
+    request_id:422,              
+    model_name:"yake_keywords__Model",       # model to be used 
+    model_parameters: {
+        "max_ngram_size":3,
+        "deduplication_threshold":0.25
+    },
+    input_items:[
+        {
+            id:"11",
+            text:"this is some text to classify", 
+            workspace_id:"timpani_meedan", 
+            target_state:"keyworded", 
+            language:"en",
+
+        },
+    ],
+    callback_url: "http://conductor/batch_reply/",
+    "retry_count": 0
+    "status_code": 200
+    "status_message":"success"
+    "results": [
+            {
+                "id": "11",     #  this id can be used to look up params from input
+                "text": "this is some text to classify",  
+                "keywords": [
+                    ["lookup",0.0020211625251083634],
+                    ["params",0.0020211625251083634]
+                ]
+            },
+           
+        ]
+}
+
+# Error response structures
```

Comment:
it would be helpful if you could elaborate more on how we want to use the http error codes in responses. if an example helps, classycat returns a whole range of different error codes that would be good for explaining/reviewing a variety of use cases.